launcher/ui/GuiUtil.cpp

```diff
@@ -51,11 +51,35 @@
 #include <settings/SettingsObject.h>
 #include "Application.h"
 
+constexpr int MaxMclogsLines = 25000;
+constexpr int InitialMclogsLines = 10000;
+constexpr int FinalMclogsLines = 14900;
+
+QString truncateLogForMclogs(const QString& logContent)
+{
+    QStringList lines = logContent.split("\n");
+    if (lines.size() > MaxMclogsLines) {
+        QString truncatedLog = lines.mid(0, InitialMclogsLines).join("\n");
+        truncatedLog +=
+            "\n\n\n\n\n\n\n\n\n\n"
+            "------------------------------------------------------------\n"
+            "----------------------- Log truncated ----------------------\n"
+            "------------------------------------------------------------\n"
+            "----- Middle portion omitted to fit mclo.gs size limits ----\n"
+            "------------------------------------------------------------\n"
+            "\n\n\n\n\n\n\n\n\n\n";
+        truncatedLog += lines.mid(lines.size() - FinalMclogsLines, FinalMclogsLines).join("\n");
```

Comment:
Oh wait I missed this: did you test this because I am fairly positive it should be
```suggestion
        truncatedLog += lines.mid(lines.size() - FinalMclogsLines).join("\n");
```

to only remove characters from the start of the string and continue to the end

Reply:
i did test it and i'm pretty sure it worked as expected, i can look at it again

Reply:
apparently it was only uploading the last 14899 lines, not 14900, this diff doesn't seem to change anything but it probably looks cleaner?

Reply:
mid behaves like substring in other languages, if you want the substring to stop at the end of the string you'd do string.length() as the second argument or omit it

Reply:
huh? https://doc.qt.io/qt-6/qlist.html#mid suggests the second argument is the length of the substring, not the index of the end of the substring

Reply:
Oh, looks like it does it to the end if you omit the second argument, but if you include it you do the length. Thanks, I guess I learned something new today!